test/mail/renderers/rfc_2822_test.exs

```diff
@@ -339,4 +339,143 @@ defmodule Mail.Renderers.RFC2822Test do
+                   }
+                 ]
+               },
+               %Mail.Message{
+                 headers: %{
+                   "content-transfer-encoding" => :base64,
+                   "content-type" => ["image/jpeg"]
+                 },
+                 parts: [],
+                 multipart: false
+               }
+             ]
+           } = message
+  end
+
+  test "will have correct part order with inline attachment" do
+    message =
+      Mail.build_multipart()
+      |> Mail.put_to("user1@example.com")
+      |> Mail.put_from({"User2", "user2@example.com"})
+      |> Mail.put_subject("Test email")
+      |> Mail.put_attachment({"tiny_jpeg.jpg", @tiny_jpeg_binary})
+      |> Mail.put_attachment({"inline_jpeg.jpg", @tiny_jpeg_binary},
+        headers: %{
+          content_id: "c_id",
+          content_type: "image/jpeg",
+          x_attachment_id: "a_id",
+          content_disposition: ["inline", filename: "filename"]
+        }
+      )
+      |> Mail.put_text("Some text")
+      |> Mail.put_html("<h1>Some HTML</h1>")
+      |> Mail.Renderers.RFC2822.reorganize()
+
+    assert %Mail.Message{
+             headers: %{"content-type" => ["multipart/mixed"]},
+             parts: [
+               %Mail.Message{
+                 headers: %{"content-type" => ["multipart/related"]},
+                 parts: [
+                   %Mail.Message{
+                     headers: %{"content-type" => ["multipart/alternative"]},
+                     parts: [
+                       %Mail.Message{
+                         headers: %{"content-type" => ["text/plain", {"charset", "UTF-8"}]},
+                         body: "Some text",
+                         parts: [],
+                         multipart: false
+                       },
+                       %Mail.Message{
+                         headers: %{"content-type" => ["text/html", {"charset", "UTF-8"}]},
+                         body: "<h1>Some HTML</h1>",
+                         parts: [],
+                         multipart: false
+                       }
+                     ]
+                   },
+                   %Mail.Message{
+                     headers: %{
+                       "content-transfer-encoding" => :base64,
+                       "content-type" => "image/jpeg",
+                       "content-id" => "c_id",
+                       "content-disposition" => ["inline", {:filename, "filename"}],
+                       "x-attachment-id" => "a_id",
+                     },
+                     parts: [],
+                     multipart: false
+                   }
+                 ]
+               },
+               %Mail.Message{
+                 headers: %{
+                   "content-transfer-encoding" => :base64,
+                   "content-type" => ["image/jpeg"]
+                 },
+                 parts: [],
+                 multipart: false
+               }
+             ]
+           } = message
+  end
 end
```

Comment:
I think we need tests for
- only text/html with only inline attachments
- only text/html with only normal attachments
- text/html and text/plain with only inline attachments
- text/html and text/plain with only normal attachments